### DISCUSSION TITLE
相同的程序运行两遍，为什么解析效果不同

### DISCUSSION THREAD

**Discussion by Tendo33** [magicpdf/Magic-PDF#299] [Q&A] [answered]
如图：
![image](https://github.com/user-attachments/assets/ccbfdee3-1807-4d82-9e8a-d7da47e4be94)

**Comment by myhloli** [comment]
这个不太合理，建议上传两次解析的auto目录打包上来给我们分析一下

**Reply by Tendo33** [reply]
我使用的demo.py，结果只有image文件夹和 md

**Reply by myhloli** [reply]
那麻烦用下命令行试试呢

**Reply by Tendo33** [reply]
现在识别的效果都是第二种了，英文识别有很大的问题，完全不是完整的单词了

**Reply by myhloli** [reply]
> 现在识别的效果都是第二种了，英文识别有很大的问题，完全不是完整的单词了

能打包上传一下auto目录吗？

**Reply by Tendo33** [reply]
https://f.ws28.cn/f/erx5g92av8q

**Reply by myhloli** [reply, marked as answer]
这篇文档比较特殊，span的排版存在一些错位和空格塞入的问题
<img width="1779" alt="image" src="https://github.com/user-attachments/assets/3e73fc81-4366-449f-b80e-71e61bfb2928">
左边是auto模式使用文字版提取逻辑得到的span，右边是ocr模式提取的span。
对于这种文档，可以在命令行中添加 --method ocr 的方式强制开启ocr解析，效果会好不少
[ocr.zip](https://github.com/user-attachments/files/16508722/ocr.zip)

**Reply by Tendo33** [reply]
有一个小问题：单论解析效果的话，开 ocr 的解析效果是不是一定是比 text 好的

**Reply by myhloli** [reply]
> 有一个小问题：单论解析效果的话，开 ocr 的解析效果是不是一定是比 text 好的

大多数情况下，解析模式用auto就可以了，一般文字版pdf直接提取里面的文字效果是比ocr要好的，但是在一小部分特殊文档上，直接提取文本质量会不达预期，这种需要手动强制开启ocr，对比一下哪种效果更好。

**Comment by myhloli** [comment]
0.10.x的auto模式对这些span不规范的pdf有优化，可以更新到最新版本或在我们的gradio demo上测试
